Assets/MixedRealityToolkit.SDK/Features/Utilities/InputRayUtils.cs

```diff
@@ -33,11 +33,18 @@ public static bool TryGetEyeGazeRay(out Ray ray)
 
             IMixedRealityEyeGazeProvider eyeGazeProvider = CoreServices.InputSystem?.EyeGazeProvider;
             if ((eyeGazeProvider == null) ||
-                !eyeGazeProvider.IsEyeGazeValid)
+                !eyeGazeProvider.IsEyeTrackingAvailable)
             { return false; }
 
-            ray.origin = eyeGazeProvider.GazeOrigin;
-            ray.direction = eyeGazeProvider.GazeDirection;
+            if (eyeGazeProvider.UseEyeTracking)
+            {
+                ray.origin = eyeGazeProvider.GazeOrigin;
+                ray.direction = eyeGazeProvider.GazeDirection;
+            }
+            else
+            {
+                ray = eyeGazeProvider.LatestEyeGaze;
```

Comment:
I do not understand why this was added. How is "eyeGazeProvider.GazeOrigin" and "...GazeDirection" different to "eyeGazeProvider.LatestEyeGaze"?

Reply:
eyeGazeProvider.GazeOrigin / GazeDirection returns current gaze origin and direction. Using this, If gaze is head based, this util would always return head ray. What we want is a way of querying the raw eye ray, even if gaze provider is not using it, the data coming from device or simulation.